### PR TITLE
Avocados

### PR DESCRIPTION
**Pull request by Shadowmech88**
avocado avocado_cut avocado_pitted
Adds avocados. Three packets of avocado seeds can now be found in the Botany seed vendor.
Before an avocado can be eaten or used in food, it must be cut in half with something sharp. This will produce one pitted half, and another half with the pit inside it. The latter must have its pit removed with something sharp before it can be used.

avocado_toast
Adds avocado toast, which requires a slice of bread, half an avocado, and 2u of salt.
avocado_burger
Adds avocado burger, which requires 5u flour, meat, and half an avocado.
avocado_soup
Adds avocado soup, which requires two avocados, a head of garlic, 5u water, 5u lime juice, and 10u cream.
avocado_milkshake
Adds avocado milkshake, which requires one avocado, one ice cream, and 10u milk.

This PR also changes the guacamole recipe to require half an avocado.

:cl:
 * rscadd: Added avocados. Before an avocado can be eaten or used, it must be cut in half with something sharp, and have its pit removed with something sharp.
 * rscadd: Added avocado toast, which requires a slice of bread, half an avocado, and 2u of salt.
 * rscadd: Added avocado burger, which requires 5u flour, meat, and half an avocado.
 * rscadd: Added avocado soup, which requires two avocados, a head of garlic, 5u water, 5u lime juice, and 10u cream.
 * rscadd: avocado milkshake, which requires one avocado, one ice cream, and 10u milk.
 * tweak: The guacamole recipe now requires half an avocado to make.

### CODE REVIEW

**Comment by Gringonius** [comment]
avolatte
Could it be?

**Comment by ghost** [comment]
Remove ice peppers from the guac recipe and add tomatoes, chilis, and black pepper reagent to it.  It is bad enough to lack cilantro and onions, but to live a minute longer under the tyranny of ice peppers is not something I will tolerate.

**Comment by carshalash** [comment]
Fuck off harse, the man found his secret recipe

maxresdefault 9

**Comment by Shadowmech88** [comment]
What *was* the original logic for using ice chilis in the recipe anyway? We have regular chilis.

**Comment by ghost** [comment]
It looks like @Kurfursten did with [this](https://github.com/vgstation-coders/vgstation13/pull/11139). 

6d76hdhy6u

You just made the list.  I strongly advise you watch your step in the future.

File: code/modules/reagents/reagent_containers/food/snacks/grown.dm [hunk 1]
```diff
@@ -945,3 +945,55 @@ var/list/special_fruits = list()
...
+/obj/item/weapon/reagent_containers/food/snacks/grown/avocado/can_consume(mob/living/carbon/eater, mob/user)
+	if(cant_eat_msg)
+		to_chat(user, "<span class='notice'>This [name][cant_eat_msg]</span>")
+	else
+		return ..()
+
+/obj/item/weapon/reagent_containers/food/snacks/grown/avocado/attackby(obj/item/weapon/W, mob/user)
+	..()
+	if(W.sharpness_flags & SHARP_BLADE)
+		if(cut)
```
Comment (Intigracy) [inline code/modules/reagents/reagent_containers/food/snacks/grown.dm]: What's preventing you from "cutting" a pitless avocado?

**Comment by Kurfursten** [comment]
@Shadowmech88 It was part of a PR to involve more mutant foods in the kitchen. Before I added ice pepper recipes, there were only two uses for that plant: a recolor of the normal chili soup and to extract frost oil for chaos donuts. The four dipping sauces were part of a set where each one required a different skill from the chef to acquire: cheesemaking (queso), mutating (guac), trading (salsa), and beekeeping (hummus). That's why it's particularly rare to see all four sauces in one kitchen. And also, as we all know, ice peppers taste like avocado.

Personally I think it's stupid to add a seed and then only one recipe for it. Every time I added a new plant I always tried to add several recipes. People adding meme plants is how we end up with Deity's cinnamon tree that no one grows. I'm always up for more botany content but I wish we could get more coverage for old plants instead of adding new plants used in barely anything. This does the opposite: removes an old plant's recipe.

**Comment by Anton-Kr** [comment]
Okay what the fuck this PR is not okay holy shit
Who the fuck uses a knife to take out the pit.

**Comment by yclatious** [comment]
People who want to get a knife inside their palms im guessing.

**Comment by Shadowmech88** [comment]
If you use a sufficiently sharp knife, it will bite into the pit with hardly any force. Less force than it would take to cut through the avocado itself.
You can then twist the seed to dislodge it and pull it out.

**Comment by Shadowmech88** [comment]
Whoops.

**Comment by Exxion** [comment]
Thanks. This is a tasteful meme.

**Comment by ghost** [comment]
>avocado toast

There goes the housing market

File: code/modules/reagents/reagent_containers/food/snacks.dm [hunk 2]
```diff
@@ -1727,6 +1727,19 @@
...
+/obj/item/weapon/reagent_containers/food/snacks/avocadosoup
+	name = "Vegetable soup"
+	desc = "May be served either hot or cold."
+	icon_state = "avocadosoup"
+	trash = /obj/item/trash/snack_bowl
+	food_flags = FOOD_LIQUID
+
+/obj/item/weapon/reagent_containers/food/snacks/avocadosoup/New()
+	..()
+	reagents.add_reagent(NUTRIMENT, 8)
```
Comment (ghost) [inline code/modules/reagents/reagent_containers/food/snacks.dm:1739]: >implying anything with avocado will be nutritious

Comment (Shadowmech88) [reply code/modules/reagents/reagent_containers/food/snacks.dm:1739]: A typical avocado serving is moderate to rich in several B vitamins and vitamin K, with good content of vitamin C, vitamin E, and potassium.
They can also be an important source of healthy fat in diets with limited access to other fatty foods such as high-fat meats and fish, or dairy products.
Additionally, avocado consumption has been shown to lower blood cholesterol levels.

Comment (Exxion) [reply code/modules/reagents/reagent_containers/food/snacks.dm:1739]: Also nutriment clearly has more to do with how full it makes you feel (meaning how much fat it has) than how nutritious it is
That's why corn oil is better nutriment than nutriment
And avocados are not lacking in fat

**Comment by Shadowmech88** [comment]
* In addition to avocado toast, avocados can now be used to make avocado burgers, avocado soup, and avocado milkshakes.

**Comment by Durgen** [comment]
pit should function as a seed

**Comment by Shadowmech88** [comment]
I thought about that, but it would mean you could skip the seed extractor altogether. Each fruit you pick would guarantee you a seed *and* you'd get to keep the fruit.
I wasn't sure if that was acceptable.

**Comment by Kurfursten** [comment]
Well at least you made a good effort to add some more recipes 👍

**Comment by RFultra** [comment]
https://www.youtube.com/watch?v=AnqPfbb0GlA

**Comment by ghost** [comment]
You can already wirecut to get infinite seeds, and you get more than one seed from the extractor anyways
I don't think anyone would care if you made the pit work as seed. Same for potatos actually

**Comment by Megamaw** [comment]
make the final bite of an avocado toast drain ten creds from your account

**Comment by Exxion** [comment]
If you're going to get into territory that memey, might as well go all the way
*Every* bite

**Comment by Megamaw** [comment]
eh, if it was per bite, i'd say more like 5 creds per bite.
either way, i'm not serious. it'd be funny for a while, but eh.

**Comment by Shadowmech88** [comment]
* When pitted, avocado halves will now drop a seed that can be planted.
* Avocado halves can no longer be fed to the seed extractor, only whole avocados can.

**Pull request merged**